src/amo/containers/App.js

```diff
@@ -1,15 +1,35 @@
+/* global window */
+
 import React, { PropTypes } from 'react';
 import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 
 import 'core/fonts/fira.scss';
 import 'amo/css/App.scss';
 import translate from 'core/i18n/translate';
-
+import { startLoginUrl } from 'core/api';
 
 export class AppBase extends React.Component {
   static propTypes = {
     children: PropTypes.node,
+    handleLogIn: PropTypes.func.isRequired,
     i18n: PropTypes.object.isRequired,
+    isAuthenticated: PropTypes.bool,
+    location: PropTypes.object.isRequired,
+  }
+
+  logIn() {
```

Comment:
I didn't want to call it `logInButtonOrLoggedInMessage` and calling it `logInButton` didn't seem right if it won't always be a button. I'll change it to `accountButton`, assuming that will make sense in the future.